src/Microsoft.AspNet.Mvc.Razor/Resources.resx

```diff
@@ -200,4 +200,7 @@
 
 @{3} "{4}, {5}"</value>
   </data>
+  <data name="LayoutHasCircularReference" xml:space="preserve">
+    <value>Possible circular layout reference detected when rendering '{0}'. The layout page '{1}' has already been rendered.</value>
```

Comment:
I was discussion this with @kichalla. There's an unorthodox scenario where you could eventually terminate the cycle:

```
// _Layout.cshtml
@{
   var node = (Node)Context.Items["node"];
   if (node != null)
   {
      Layout = "_Layout";
     Context.Items["node"] = node.Parent;
   }
}
@RenderBody()

```